File: spotifyarch/src/main/java/com/timothy/spotifyarchitecture/retrofit/models/PlaylistTrack.java
package com.timothy.spotifyarchitecture.retrofit.models;

import android.arch.persistence.room.Embedded;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.ForeignKey;

/**
 * <a href="https://developer.spotify.com/web-api/object-model/#playlist-track-object">Playlist track object model</a>
 */
@Entity(primaryKeys = {"track", "added_by"},
        tableName = "playlist_tracks",
        foreignKeys = @ForeignKey(entity = Playlist.class,
                                    parentColumns = "playlist_id",
                                    childColumns = "track",
                                    onDelete = ForeignKey.CASCADE))
public class PlaylistTrack {

    public String added_at;
    @Embedded
    public UserPublic added_by;
    @Embedded()
    public Track track;
    public Boolean is_local;

    public PlaylistTrack() {
    }

}

File: spotifyarch/src/main/java/com/timothy/spotifyarchitecture/retrofit/models/PlaylistTrack.kt
package com.timothy.spotifyarchitecture.retrofit.models

import android.arch.persistence.room.Entity
import android.arch.persistence.room.ForeignKey

/**
 * [Playlist track object model](https://developer.spotify.com/web-api/object-model/#playlist-track-object)
 */
@Entity(primaryKeys = arrayOf("track", "added_by"), tableName = "playlist_tracks", foreignKeys = arrayOf(ForeignKey(entity = Playlist::class, parentColumns = arrayOf("playlist_id"), childColumns = arrayOf("track"), onDelete = ForeignKey.CASCADE)))
class PlaylistTrack(val added_at: String? = "", val added_by: UserPublic? = UserPublic(), val track: Track = Track(), val is_local: Boolean = false)